### PR TITLE
MaterializeCPULaunch configurations before bufferization

### PR DESCRIPTION
**Pull request by asaadaldien**
- Move materlialize configs to be the first thing to run.
- Move distributed affine.min and other canoncalization passes to run as
post materialization step.

Depends on #5369

### CODE REVIEW

File: iree/compiler/Conversion/LinalgToLLVM/Passes.cpp [hunk 1]
```diff
@@ -77,13 +68,12 @@ void addLinalgToLLVMPasses(OpPassManager &passManager,
...
-  OpPassManager &nestedModulePM = passManager.nest<ModuleOp>();
-
-  nestedModulePM.addPass(createInlinerPass());
-
-  // HLO -> Linalg on buffers.
   if (options.usingLinalgOnTensors) {
-    nestedModulePM.addNestedPass<FuncOp>(createLinalgVectorizePass());
+    passManager.addPass(createMaterializeCPULaunchConfigurationPass());
+    OpPassManager &nestedModulePM = passManager.nest<ModuleOp>();
+    nestedModulePM.addPass(createInlinerPass());
```
Comment (MaheshRavishankar) [inline iree/compiler/Conversion/LinalgToLLVM/Passes.cpp:74]: Probably better to move the inline pass before. Any reason why you moved it after?

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]

File: iree/compiler/Conversion/LinalgToLLVM/Passes.cpp (hunk 1 above)
Comment (asaadaldien) [reply iree/compiler/Conversion/LinalgToLLVM/Passes.cpp:74]: As we discussed before we want to get these constant materialized as early as possible and substituting the tiling constants right after the backend is picked.

Comment (MaheshRavishankar) [reply iree/compiler/Conversion/LinalgToLLVM/Passes.cpp:74]: Yes, I remember. Just saying maybe do this after inlining. Actually probably dont need inlining on the Linalg on tensors path.

Comment (asaadaldien) [reply iree/compiler/Conversion/LinalgToLLVM/Passes.cpp:74]: Yeah, good point! looking at the IR dump of mobile bert, seems like the inliner is totally no-op here. will remove it

Comment (asaadaldien) [reply iree/compiler/Conversion/LinalgToLLVM/Passes.cpp:74]: I can send a followup PR to remove the `Ininer`

**Review by MaheshRavishankar — APPROVED** [review]

**Pull request merged**